extension/source/background/AggregatorController.ts

```diff
@@ -61,24 +65,43 @@ export default class AggregatorController {
         this.ethersProvider,
       );
 
-      const nonce = (await wallet.Nonce()).toString();
-      const bundle = await wallet.sign({ nonce, actions });
-
-      const aggregatorUrl =
-        this.preferredAggregators[providerId] ?? AGGREGATOR_URL;
-      const agg = new Aggregator(aggregatorUrl);
-      const result = await agg.add(bundle);
-
-      assert(!('failures' in result), () => new Error(JSON.stringify(result)));
+      const userAction = await this.transactionsController
+        .InternalRpc()
+        .requestTransaction(...params);
+
+      if (userAction === TransactionStatus.APPROVED) {
+        // const nonce = (await wallet.Nonce()).toString();
+        const nonce = (
+          await BlsWalletWrapper.Nonce(
+            await wallet.PublicKey(),
+            NETWORK_CONFIG.addresses.verificationGateway,
+            this.ethersProvider,
+          )
+        ).toString();
+        const bundle = await wallet.sign({ nonce, actions });
+
+        const aggregatorUrl =
+          this.preferredAggregators[providerId] ?? AGGREGATOR_URL;
+        const agg = new Aggregator(aggregatorUrl);
+        const result = await agg.add(bundle);
+
+        assert(
+          !('failures' in result),
+          () => new Error(JSON.stringify(result)),
+        );
+
+        this.knownTransactions[result.hash] = {
+          ...params[0],
+          nonce,
+          value: params[0].value ?? '0',
+          aggregatorUrl,
+        };
```

Comment:
We can use TransactionsController instead of this object. Did not remove it yet to keep the scope of this PR small. Created an issue to track this https://github.com/web3well/bls-wallet/issues/256